pyanaconda/ui/lib/payload.py

```diff
@@ -59,3 +60,58 @@ def get_payload(payload_type=PAYLOAD_TYPE_DNF):
 
     # Or create a new payload.
     return create_payload(payload_type)
+
+
+def create_source(source_type):
+    """Create a source.
+
+    :param source_type: a source type
+    :return: a DBus proxy of a source
+    """
+    payloads_proxy = PAYLOADS.get_proxy()
+    object_path = payloads_proxy.CreateSource(source_type)
+    return PAYLOADS.get_proxy(object_path)
+
+
+def set_source(payload_proxy, source_proxy):
+    """Attach the source to the payload.
+
+    :param payload_proxy: a DBus proxy of a payload
+    :param source_proxy: a DBus proxy of a source
+    """
+    object_path = get_object_path(source_proxy)
+    payload_proxy.SetSources([object_path])
+
+
+def get_source(payload_proxy, default_source_type=None):
+    """Get a source of the given payload.
+
+    If the payload has one or more sources, return the first one.
```

Comment:
What's that good for? I get something, and don't know if there is more?

Reply:
It is a temporary workaround, because the DBus API is designed to work with multiple sources, but UI doesn't support them, so it will ignore other sources for now.

Reply:
Ah, ok.